posts/2020/08/versioned-hdf5-performance.md

```diff
@@ -0,0 +1,138 @@
+
+## I/O Performance for Versioned-HDF5 Files
+
+First, we'll look at the time required to read data from all versions in a file, sequentially. To keep this test short, we’ll only analyze the tests using **no compression**, and **chunk size** 16384 for the one-dimensional datasets and 256 for the two-dimensional dataset in `test_mostly_appends_dense`.
+
+Plotting with respect to the number of versions, we get the following:
+
+![](/images/versioned-hdf5-performance/seqread.png)
+
+As expected, read times increase for files with a larger number of versions, but the growth is close to linear in all cases except for `test_mostly_appends_dense`, where the large array sizes explain the larger read times.
+
+Next, we’ll compare the times necessary to read the latest version on each file. Because of how VersionedHDF5 is designed, this is the same as reading the last array stored in the HDF5 file. For each test, we made 20 different reads of the latest version in order to eliminate fluctuations generated by background processes or the OS (solid lines). We also compare these read times with the time required to read an unversioned file (which only stored the latest version - dashed black line).
+
+![](/images/versioned-hdf5-performance/latestread.png)
+
+In this case, we can see that on average, reading the latest version on a VersionedHDF5File is ~5x-10x slower than reading an unversioned file. Also, the time required to read the latest version from a Versioned HDF5 file increases modestly with the number of versions stored in the file, except when the size of the array increases significantly with each new version.
+
+## Summary
+
+Overall, the worst performance was observed for tests with large array sizes. This seems to show that the file sizes and IO performance of Versioned HDF5 files are significantly affected by the size of the unique datasets stored in each file, which is to be expected. 
+
+The results presented here show that the largest impact on IO performance and storage is in fact explained by the size of the datasets stored in the file, and not significantly reduced by the Versioned HDF5 abstraction.
+
+Next steps
+----------
+
+This is the second post in a series about the Versioned HDF5 library. Next,
+we'll discuss the the design of the Versioned HDF5 library.
+
+`versioned-hdf5 1.0` has recently been released, and is available on PyPI and conda-forge. You can install it with
+
+```py
+conda install -c conda-forge versioned-hdf5
+```
+
+The development is on [GitHub](https://github.com/deshaw/versioned-hdf5).
```

Comment:
I propose a slight rephrase: "Development for `versioned-hdf5` happens on [GitHub]"